packages/node_modules/@webex/plugin-meetings/test/unit/spec/meetings/index.js

```diff
@@ -0,0 +1,235 @@
+              assert.calledWith(spark.meetings.request.getActiveMeetings);
+              assert.calledWith(MeetingCollection.getByKey, 'locusUrl', 'test');
+              assert.calledWith(onFullLocus, {url: 'test'});
+            });
+          });
+          describe('when meeting is not returned', () => {
+            let initialSetup;
+            beforeEach(() => {
+              initialSetup = sinon.stub().returns(true);
+              MeetingCollection.getByKey = sinon.stub().returns(null);
+              spark.meetings.create = sinon.stub().returns(Promise.resolve({locusInfo: {initialSetup}}));
+            });
+            it('tests the sync meeting calls for not existing meeting', async () => {
+              await spark.meetings.syncMeetings();
+              assert.calledOnce(spark.meetings.request.getActiveMeetings);
+              assert.calledOnce(MeetingCollection.getByKey);
+              assert.calledOnce(initialSetup);
+              assert.calledOnce(spark.meetings.create);
+              assert.calledWith(spark.meetings.request.getActiveMeetings);
+              assert.calledWith(MeetingCollection.getByKey, 'locusUrl', 'test');
+              assert.calledWith(spark.meetings.create, {url: 'test'}, 'LOCUS_ID');
+              assert.calledWith(initialSetup, {url: 'test'});
+            });
+          });
+        });
+      });
+      describe('#create', () => {
+        beforeEach(() => {
+          spark.meetings.createMeeting = sinon.stub().returns(Promise.resolve(true));
+        });
+        it('calls createMeeting and returns its promise', async () => {
+          const create = spark.meetings.create('test', 'test');
+          assert.exists(create.then);
+          await create;
+          assert.calledOnce(spark.meetings.createMeeting);
+          assert.calledWith(spark.meetings.createMeeting, 'test', 'test');
+        });
+      });
+      describe('#createMeeting', () => {
+        beforeEach(() => {
+          spark.internal.device.userId = 'test';
+          spark.internal.device.url = 'test';
+          MeetingCollection.set = sinon.stub().returns(true);
+          MeetingsUtil.getMeetingAddedType = sinon.stub().returns('test');
+          MeetingsUtil.extractDestination = sinon.stub().returns('test');
+        });
+        describe('successful MeetingInfo.#fetchMeetingInfo', () => {
+          beforeEach(() => {
+            spark.meetings.meetingInfo.fetchMeetingInfo = sinon.stub().returns(Promise.resolve(true));
+          });
+          it('creates the meeting from a successful meeting info fetch', async () => {
+            const meeting = await spark.meetings.createMeeting('test', 'test');
+            assert.instanceOf(meeting, Meeting, 'createMeeting should eventually resolve to a Meeting Object');
+            assert.calledOnce(spark.meetings.meetingInfo.fetchMeetingInfo);
+            assert.calledOnce(MeetingsUtil.extractDestination);
+            assert.calledOnce(MeetingsUtil.getMeetingAddedType);
```

Comment:
I don't know if we need to get this granular on asserting individual steps inside of `create`. The most important thing is we assert on what the expected return value is (and that it resolves/rejects a promise).

Reply:
I'll add 2 tests, one detailed and one contract based.